### PR TITLE
EOS-21226: Fixed ut/sm.c

### PR DESCRIPTION
**Pull request by shashank-parulekar**
Changed the size of permissive_trans array from 256 elements to 225 elements to
make the ut run without crashing.

Signed-off-by: Shashank Parulekar <shashank.parulekar@seagate.com>

### CODE REVIEW

File: sm/ut/sm.c [hunk 1]
```diff
@@ -628,9 +628,9 @@ static void ast_wait(void)
 	m0_mutex_fini(&wait_guard);
 }
 
-static struct m0_sm_state_descr permissive_states[16] = {};
+static struct m0_sm_state_descr permissive_states[15] = {};
```
Comment (NikhilBirgade) [inline sm/ut/sm.c:631]: It will better to have a #define.

File: sm/ut/sm.c [hunk 2]
```diff
@@ -628,9 +628,9 @@ static void ast_wait(void)
...
 
-static struct m0_sm_trans_descr permissive_trans[256] = {};
+static struct m0_sm_trans_descr permissive_trans[225] = {};
```
Comment (NikhilBirgade) [inline sm/ut/sm.c:633]: It will better to have a #define.

**Review by ydb242 — APPROVED** [review]
Barring nikhil's comment, the PR looks good!

File: sm/ut/sm.c (hunk 1 above)
Comment (shashank-parulekar) [reply sm/ut/sm.c:631]: sure. let me use a define instead

File: sm/ut/sm.c (hunk 2 above)
Comment (shashank-parulekar) [reply sm/ut/sm.c:633]: sure. let me use a define instead

**Review by NikhilBirgade — APPROVED** [review]

**Review by RadhaGulhane13 — APPROVED** [review]

**Pull request merged**